### PR TITLE
usm: mongo: Added mongo classification

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?

Adds mongo classification code to NPM / USM

### Motivation

Support more protocols in classification

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by bkabrda — APPROVED** [review]

File: pkg/network/ebpf/c/protocols/protocol-classification-helpers.h [hunk 1]
```diff
@@ -236,6 +285,8 @@ static __always_inline void classify_protocol(protocol_t *protocol, const char *
         *protocol = PROTOCOL_AMQP;
     } else if (is_redis(buf, size)) {
         *protocol = PROTOCOL_REDIS;
+    } else if (is_mongo(tup, buf, size)) {
```
Comment (val06) [inline pkg/network/ebpf/c/protocols/protocol-classification-helpers.h:288]: do we really must use connection tuple as an identifier for mongo classification?

If possible we better to stick to the same interface for all protocols and drop that additional parameter you added to `classify_protocol` function

File: pkg/network/tracer/tracer_classification_test.go [hunk 2]
```diff
@@ -537,6 +539,139 @@ func testProtocolClassification(t *testing.T, cfg *config.Config, clientHost, ta
 			teardown:   defaultTeardown,
 			validation: validateProtocolConnection,
 		},
+		{
```
Comment (val06) [inline pkg/network/tracer/tracer_classification_test.go:542]: what about some cases to ensure we don't identify false-positives?

**Review by val06 — CHANGES_REQUESTED** [review]
cred

File: pkg/network/ebpf/c/protocols/protocol-classification-helpers.h (hunk 1 above)
Comment (guyarb) [reply pkg/network/ebpf/c/protocols/protocol-classification-helpers.h:288]: we must correlate between request and response as response classification is very "soft"

File: pkg/network/tracer/tracer_classification_test.go (hunk 2 above)
Comment (guyarb) [reply pkg/network/tracer/tracer_classification_test.go:542]: do you have an example? I can create random buffers but that's not good enough
the real scenario we wished to check here is multiple different protocols (even not supported protocols) and ensure we do not classify them, but it wont guarantee success

**Review by val06 — CHANGES_REQUESTED** [review]
cred

File: pkg/network/ebpf/c/protocols/protocol-classification-helpers.h [hunk 3]
```diff
@@ -237,34 +256,31 @@ static __always_inline bool is_mongo(conn_tuple_t *tup, const char *buf, __u32 s
...
     }
 
-    mongo_key key = {};
-    key.tup = *tup;
-
     switch (header.op_code) {
+    case MONGO_OP_UPDATE:
+    case MONGO_OP_INSERT:
+    case MONGO_OP_DELETE:
+        // If the response_to is not 0, than it is not a valid mongo request by the RFC.
```
Comment (val06) [inline pkg/network/ebpf/c/protocols/protocol-classification-helpers.h:263]: then

File: pkg/network/ebpf/c/protocols/protocol-classification-helpers.h [hunk 4]
```diff
@@ -237,34 +256,31 @@ static __always_inline bool is_mongo(conn_tuple_t *tup, const char *buf, __u32 s
...
-    case MONGO_OP_DELETE:
-    case MONGO_OP_KILL_CURSORS:
+        if (header.response_to == 0) {
+            mongo_handle_request(tup, header.request_id);
+            return true;
+        }
+        return false;
     case MONGO_OP_COMPRESSED:
     case MONGO_OP_MSG:
         // If the response_to is not 0, than it is not a valid mongo request by the RFC.
```
Comment (val06) [inline pkg/network/ebpf/c/protocols/protocol-classification-helpers.h:278]: then

**Review by val06 — CHANGES_REQUESTED** [review]
cred

**Review by val06 — APPROVED** [review]

**Pull request merged**